### PR TITLE
Add `TypeKind::Qualified` and remove `Type::is_const`

### PR DESCRIPTION
**Pull request by pvdrz**

### CODE REVIEW

File: bindgen/ir/context.rs [hunk 1]
```diff
@@ -1764,12 +1764,9 @@ If you encounter an error missing from this list, please file an issue or a PR!"
...
         let name = if name.is_empty() { None } else { Some(name) };
-        let ty = Type::new(
-            name,
-            ty.fallible_layout(self).ok(),
-            type_kind,
-            ty.is_const(),
-        );
+        let is_const = ty.is_const();
+        let layout = ty.fallible_layout(self).ok();
+        let ty = Type::new(name.clone(), layout.clone(), type_kind);
```
Comment (emilio) [inline bindgen/ir/context.rs:1769]: nit: just `layout` as per the clippy thing.

File: bindgen/ir/ty.rs [hunk 2]
```diff
@@ -1214,7 +1216,32 @@ impl Type {
                 ty.elem_type()
                     .map_or(false, |element| element.is_const()));
 
-        let ty = Type::new(name, layout, kind, is_const);
+        let mut ty = Type::new(name.clone(), layout.clone(), kind);
+
+        if is_const {
```
Comment (emilio) [inline bindgen/ir/ty.rs:1221]: So we should be able to abstract this into a `maybe_qualify` helper or something. Otherwise it feels very clunky, wdyt?

**Review by emilio — CHANGES_REQUESTED** [review]
So, generally it looks fine. I expected it to simplify some other places a bit more, specially since we still need to store the type name and layout in two places...

So I'm not convinced this is _quite_ worth it anymore, but if we want to land this at least we should factor out the qualification stuff to its own function so it isn't so repetitive...

**Review by emilio — CHANGES_REQUESTED** [review]

File: bindgen/ir/ty.rs [hunk 3]
```diff
@@ -1214,10 +1217,69 @@ impl Type {
...
+                    is_const: true,
+                },
+            );
+        }
+
         // TODO: maybe declaration.canonical()?
         Ok(ParseResult::New(ty, Some(cursor.canonical())))
     }
+
+    pub(crate) fn maybe_qualify(
```
[the remark this answers is not in the document]
Comment (pvdrz) [reply bindgen/ir/ty.rs:1252]: You're right. I documented the function and avoided returning the id.

**Pull request closed without merging**